core/src/main/java/org/apache/calcite/runtime/ProjectionTransformer.java

```diff
@@ -40,6 +40,14 @@
 
 /**
  * Transforms the projection of a geometry.
+ *
+ * This class uses Proj4J to transform the projection of a geometry
+ * and should not be used beyond the scope of the Spatial Type Extensions.
+ * Proj4J is released under the Apache License 2.0, however, it also uses the EPSG dataset,
+ * which has restricting <a href="https://epsg.org/terms-of-use.html">terms of use</a>.
+ * As a result, Proj4J is not suitable for inclusion in Apache Calcite
+ * and this class will throw {@code ClassNotFoundException}s
+ * if Proj4J is not added to the classpath by the user.
```

Comment:
Would we need to add a similar warning in the web documentation (`reference.md`) in the spatial section?

Reply:
This is a good idea, I added notices in `reference.md` and `spatial.md`.